### PR TITLE
Master backmerge into Develop

### PR DESCRIPTION
**Pull request by revanth0212**
## Description

Backmerging `master` branch into `develop` branch.

## Related Issue
Closes PWA-1333

### Verification Stakeholders
@jimbo @dpatil-magento 

### Verification Steps
A smoke test on the deployed app. Nothing should break, everything should fine.

## Checklist
* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Comment by revanth0212** [comment]
Did a smoke test and everything looks fine.

**Review by tjwiebell — APPROVED** [review]
Changeset looks straight forward, LGTM 👍

File: packages/venia-ui/lib/RootComponents/Product/product.js [hunk 1]
```diff
@@ -23,6 +23,8 @@ const Product = () => {
 
     const { error, loading, product } = talonProps;
 
+    if (loading && !product) return fullPageLoadingIndicator;
+    if (error && !product) return <ErrorView />;
     if (!product) {
         if (loading) return fullPageLoadingIndicator;
         else if (error) {
```
Comment (tjwiebell) [inline packages/venia-ui/lib/RootComponents/Product/product.js]: Theses two gates are unreachable now. I think we should have just accepted `develop` in this resolution, eg.

```
if (loading && !product) return fullPageLoadingIndicator;
if (error && !product) return <ErrorView />;
if (!product) { // OOS Message }
```

**Review by tjwiebell — CHANGES_REQUESTED** [review]
Didn't catch the conflict resolutions snuck into that merge commit, did have some minor feedback that needs addressed.

File: packages/venia-ui/lib/RootComponents/Product/product.js (hunk 1 above)
Comment (revanth0212) [reply packages/venia-ui/lib/RootComponents/Product/product.js]: You are right @tjwiebell. @sirugh what do you think should be the result if there is an error?

In `develop` we are rendering

```jsx
<ErrorView />
```

And in `master` we are rendering 

```jsx
<div>
     <FormattedMessage
            id={'product.errorFetch'}
            defaultMessage={'Data Fetch Error'}
       />
</div>
```

Comment (sirugh) [reply packages/venia-ui/lib/RootComponents/Product/product.js]: I think you should use what is in `develop`, as @tjwiebell suggested.

**Review by tjwiebell — APPROVED** [review]

**Comment by dpatil-magento** [comment]
@revanth0212 looks good, just unhide these links now https://github.com/magento/pwa-studio/pull/2918

**Comment by revanth0212** [comment]
> @revanth0212 looks good, just unhide these links now #2918

@dpatil-magento done 👍

**Review by tjwiebell — APPROVED** [review]

**Comment by dpatil-magento** [comment]
QA Approved.

**Pull request merged**